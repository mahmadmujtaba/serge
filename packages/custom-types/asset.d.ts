import Perception from './perception'
import PlannedTurn from './planned-turn'

export interface Asset {
  name: string,
  uniqid: string,
  position: string,
  perceptions: Perception[],
  plannedTurns?: PlannedTurn[]
}

export default Asset
