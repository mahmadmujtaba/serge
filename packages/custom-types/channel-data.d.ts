import Participant from './participant'
import PlayerUiMessageTypes from './player-ui-message'

export default interface ChannelData {
  // TODO: change it to use name
  uniqid?: string,
  observing?: boolean,
  templates?: Array<any>,
  messages?: Array<PlayerUiMessageTypes>,
  name: string,
  participants: Array<Participant>
  unreadMessageCount?: number
  // TODO: dynamically retrieve force icons and colors,
  // don't store them in the database
  forceIcons?: Array<any>
  forceColors?: Array<string>
}
