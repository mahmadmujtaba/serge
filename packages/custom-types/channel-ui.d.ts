import Participant, { CoreParticipant } from './participant'
import { MessageChannel } from './message'
import { TemplateBody } from './template'
import { SpecialChannelTypes } from '@serge/config'
import { CollabOptions } from './channel-data'

/** description of channel, as used in game UI */
export default interface ChannelData {
  readonly uniqid: string,
  observing?: boolean,
  templates?: TemplateBody[],
  messages?: Array<MessageChannel>,
  name: string,
  participants: Array<Participant>
  v3participants?: Array<CoreParticipant>
  unreadMessageCount?: number
  // TODO: dynamically retrieve force icons and colors,
  // don't store them in the database
  forceIcons?: Array<any>
  forceColors?: Array<string>
  format?: SpecialChannelTypes
  /** one of the v3 channel type constants */
  channelType: string,
  collabOptions? : CollabOptions
}
