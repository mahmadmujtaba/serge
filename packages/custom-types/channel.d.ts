import { ChannelTypes } from './channel-data'

export default interface Channel {
  name: string,
  channels: Array<ChannelTypes>,
  selectedChannel: string,
  complete: boolean,
  dirty: boolean
}