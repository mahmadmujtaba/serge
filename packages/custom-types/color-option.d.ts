import ForceData from './force-data'

export default interface ColorOption {
  name: string,
  colour: string
}