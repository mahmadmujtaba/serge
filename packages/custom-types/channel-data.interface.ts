import Participant from './participant.interface'

export default interface ChannelData {
  name: string,
  participants: Array<Participant>
}