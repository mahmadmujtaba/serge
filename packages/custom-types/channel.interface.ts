import ChannelData from './channel-data.interface'

export default interface Channel {
  name: string,
  channels?: Array<ChannelData>,
  selectedChannel: string,
  complete: boolean,
  dirty: boolean
}