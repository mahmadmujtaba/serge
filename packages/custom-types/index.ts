import ChannelData from './channel-data.interface'
import Channel from './channel.interface'
import ForceData from './force-data.interface'
import Force from './force.interface'
import MarkerPerception from './marker-perception.interface'
import Participant from './participant.interface'
import PlatformTypeData from './platform-type-data.interface'
import PlatformType from './platform-type.interface'
import Role from './role.interface'
import States from './states.interface'
import Template from './template.interface'
import WargameData from './wargame-data.interface'
import WargameList from './wargame-list.interface'
import WargameOverview from './wargame-overview.interface'
import Wargame from './wargame.interface'

export {
  ChannelData,
  Channel,
  ForceData,
  Force,
  MarkerPerception,
  Participant,
  PlatformTypeData,
  PlatformType,
  Role,
  States,
  Template,
  WargameData,
  WargameList,
  WargameOverview,
  Wargame
}