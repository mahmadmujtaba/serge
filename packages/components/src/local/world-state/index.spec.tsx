/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import WorldState from './index'
import { RouteStore } from '@serge/custom-types'
const store: RouteStore = { routes: [] }

it('WorldState renders correctly', () => {
  const tree = renderer
    .create(<WorldState store={store} name="test" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
