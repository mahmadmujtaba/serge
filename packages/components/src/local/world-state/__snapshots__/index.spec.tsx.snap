// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorldState renders correctly 1`] = `
<div
  className="world-state"
>
  <h2>
    test
  </h2>
  <ul>
    <li>
      == [Under my control] ==
    </li>
    <li>
      == [Not under my control] ==
    </li>
  </ul>
</div>
`;
